src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.cs

```diff
@@ -39,7 +40,16 @@ protected AbstractSyntaxClassificationService()
         {
             try
             {
-                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                SemanticModel semanticModel;
+                if (document.Project.Solution.Services.GetRequiredService<IWorkspaceConfigurationService>().Options.DisableNullableAnalysisInClassification)
+                {
+                    semanticModel = await document.GetRequiredNullableDisabledSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                }
```

Comment:
Tagging @CyrusNajmabadi on this bit, since we were quite surprised we're not using speculative semantic models here.